include/fmt/format.h

```diff
@@ -3198,17 +3198,24 @@ FMT_CONSTEXPR basic_string_view<Char> compile_string_to_view(
   return {s.data(), s.size()};
 }
 
-#define FMT_STRING_IMPL(s, base)                                  \
-  [] {                                                            \
-    /* Use a macro-like name to avoid shadowing warnings. */      \
-    struct FMT_COMPILE_STRING : base {                            \
-      using char_type = fmt::remove_cvref_t<decltype(s[0])>;      \
-      FMT_MAYBE_UNUSED FMT_CONSTEXPR                              \
-      operator fmt::basic_string_view<char_type>() const {        \
-        return fmt::detail::compile_string_to_view<char_type>(s); \
-      }                                                           \
-    };                                                            \
-    return FMT_COMPILE_STRING();                                  \
+#if FMT_GCC_VERSION
+#  define FMT_COMPILE_STRING_VISIBILITY_ATTRIBUTE \
+    __attribute__((visibility("hidden")))
+#else
+#  define FMT_COMPILE_STRING_VISIBILITY_ATTRIBUTE
+#endif
+
+#define FMT_STRING_IMPL(s, base)                                               \
+  [] {                                                                         \
+    /* Use a macro-like name to avoid shadowing warnings. */                   \
+    struct FMT_COMPILE_STRING_VISIBILITY_ATTRIBUTE FMT_COMPILE_STRING : base { \
```

Comment:
Also please add a short comment explaining why we need visibility hidden here.

Reply:
done, hope this will be enough